packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSlider.ts

```diff
@@ -31,6 +49,35 @@ export const useColorSlider_unstable = (
     thumb,
   } = props;
 
+  const hsvColor = color || colorFromContext;
+
+  const [currentValue, setCurrentValue] = useControllableState({
+    defaultState: props.defaultColor?.h,
+    state: hsvColor?.h,
+    initialState: 0,
+  });
+  const clampedValue = clamp(currentValue, MIN, MAX);
+  const valuePercent = getPercent(clampedValue, MIN, MAX);
+
+  const inputOnChange = input?.onChange;
+
+  const _onChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(event => {
+    const newValue = Number(event.target.value);
+    const newColor = { ...hsvColor, h: newValue };
+    setCurrentValue(newValue);
+    inputOnChange?.(event);
+    onChange?.(event, { type: 'change', event, color: newColor });
```

Comment:
Could you please confirm if it is expected behavior for the `onChangeFromContext` to be called twice when the component `onChange` is not provided?

Reply:
thanks for catching that. I removed duplicated event